contracts/infrastructure/dapp/paraswap/CurveFilter.sol

```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2021  Argent Labs Ltd. <https://argent.xyz>
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+// SPDX-License-Identifier: GPL-3.0-only
+pragma solidity ^0.8.3;
+
+import "../BaseFilter.sol";
+
+contract CurveFilter is BaseFilter {
+
+    bytes4 private constant EXCHANGE = bytes4(keccak256("exchange(int128,int128,uint256,uint256)"));
+    bytes4 private constant EXCHANGE_UNDERLYING = bytes4(keccak256("exchange_underlying(int128,int128,uint256,uint256)"));
+    bytes4 private constant ERC20_APPROVE = bytes4(keccak256("approve(address,uint256)"));
+
+    function isValid(address /*_wallet*/, address /*_spender*/, address /*_to*/, bytes calldata _data) external view override returns (bool valid) {
+        // disable ETH transfer
+        if (_data.length < 4) {
+            return false;
+        }
+
+        bytes4 methodId = getMethod(_data);
+
+        return methodId == ERC20_APPROVE || methodId == EXCHANGE || methodId == EXCHANGE_UNDERLYING;
```

Comment:
As mentioned earlier I would prefer to separate the cases where the target of the `_data` is the pool (`_to == _spender`) and where it is an ERC20 (`_to != _spender`) to make the filter logic more explicit.

Reply:
Sorry, did not realise it was already addressed.